### PR TITLE
feat : add filter functionality

### PR DESCRIPTION
**Pull request by Omkar-Chaskar**
added filter with context hook, reducer , mock API

### CODE REVIEW

**Review by Omkar-Chaskar — COMMENTED** [review]
deploy preview : https://deploy-preview-2--ratha-shop.netlify.app/

**Comment by ankit9015** [comment]
Preview is not working

File: src/Components/Product.js [hunk 1]
```diff
@@ -1,222 +1,48 @@
...
-                </div>
-              </div>
-              <div className="card__secondary text-center card__secondary-vertical">
-                <p className="p1"> 
-                <p className="card__subtitle card-price bold p3">
-              &#8377;2000 {" "}<span className="card-cut-price">&#8377;3999</span>
-            </p>
+                <p className="card__title">
+                  {products.subtitle} <br />
+                  {products.description}
```
Comment (amirnathani9) [inline src/Components/Product.js:43]: u can use destructuring here

File: src/Components/Product.js [hunk 2]
```diff
@@ -1,222 +1,48 @@
...
+                  <br />
```
Comment (amirnathani9) [inline src/Components/Product.js:44]: try to not use br tags

File: src/index.js [hunk 3]
```diff
@@ -3,12 +3,21 @@ import ReactDOM from "react-dom";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
+import { FilterProvider } from "./context/filter-provider";
+import { ProductProvider } from "./context/product-provider";
+import { CategoryProvider } from "./context/category-provider";
```
Comment (amirnathani9) [inline src/index.js:8]: u can also make the index.js file in context folder to export the components or pages or providers

**Review by amirnathani9 — APPROVED** [review]
Some changes suggested above and one thing if the pr gets to much big try to divide it into smaller ones. Try to keep max 500 lines of code

And things i like are
- file structuring
- code is clean and easy to read

File: src/context/product-provider.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  import axios from "axios";
+  
+  const ProductContext = createContext();
+  const useProduct = () => useContext(ProductContext);
+  
+  const ProductProvider = ({ children }) => {
+
+    const [productsList, setProductsList] = useState([]);
+    const productApiCall = async () => {
+      const response = await axios.get("/api/products");
```
Comment (rahulmistry751) [inline src/context/product-provider.js:16]: try-catch should be added for error handling of API fetching

**Review by rahulmistry751 — APPROVED** [review]

File: src/context/product-provider.js (hunk 4 above)
Comment (Omkar-Chaskar) [reply src/context/product-provider.js:16]: i will do it in next pr

**Pull request merged**